Sb/cpu inference

bes-dev opened the pull request:
Added model's device support to samplers to support CPU/TPU/etc. Inference (not only CUDA device)

bes-dev commented:
@rromb please look at PR

leszekhanusz commented:
Sorry but this PR does not make any sense and is not working.

* adding arguments to methods which does not exist
* using `device` in places where it is not defined
* and missing a lot of `.cuda()` calls
* only modifying `img2img.py` and not other scripts

bes-dev commented:
@leszekhanusz so, we replaced hardcoded .cuda() calls with generic .to(device) to make inferences using not only GPU devices. It works when we run the script on the PC without CUDA capable device. We didn't change the default behavior of existing scripts, so we changed just scripts that use SMPL.make_scheduler() call. We tested this PR using PC without CUDA device and it works for me.

plaksivayatryapka commented:
Still one hardcoded .cuda() call left unfixed in text2img.py. Made minor changes, after them it worked on the CPU, requested the pull to your fork. One picture 512x512 is generated in 8 minutes (AMD 3600x), takes up to 8 GB in RAM. That is, about 20 times slower than at 3060. But it's better than nothing for those who do not have a GPU.

shankarsivarajan commented:
I had to add the line `torch.cuda.is_available = lambda : False` to the top to use my CPU, but `txt2img.py` now works for me. Thanks!

AlexmJoh commented:
Anyone know how to get around this?
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cpu and cuda:0! (when checking argument for argument index in method wrapper__index_select)
